model/state_model_recurse.jl

```diff
@@ -0,0 +1,760 @@
+# labor income        
+        PL[r] * sum(le0[r,s] for s in sectors)
+# capital income        
+            +sum((haskey(RK.lookup[1], (r, s)) ? RK[(r,s)] : 1.0) * (ks_n[r,s]+ks_s[r,s]) for s in sectors)
+            +sum((haskey(RKX.lookup[1], (r, s)) ? RKX[(r,s)] : 1.0) * ks_x[r,s] for s in sectors)
+        #+ sum((haskey(PK.lookup[1], (r, s)) ? PK[(r,s)] : 1.0) * kd0[r,s] for s in sectors)
+# provision of household supply          
+        + sum( (haskey(PY.lookup[1], (r, g)) ? PY[(r, g)] : 1.0) * yh0[r,g] for g in goods)
+# revenue or costs of foreign exchange including household adjustment   
+        + PFX * (bopdef0[r] + hhadj[r])
+# government and investment provision        
+        - sum((haskey(PA.lookup[1], (r, g)) ? PA[(r, g)] : 1.0) * (g0[r,g] + i0[r,g]) for g in goods)
+# import taxes - assumes lumpsum recycling
+        + sum((haskey(A.lookup[1], (r, g)) ? A[(r, g)] : 1.) * MD[r,g] * PFX * tm[r,g] for g in goods if (a_set[r,g] != 0))
+# taxes on intermediate demand - assumes lumpsum recycling
+        + sum((haskey(A.lookup[1], (r, g)) ? A[(r, g)] : 1.) * a0[r,g]*(haskey(PA.lookup[1], (r, g)) ? PA[(r, g)] : 1.0)*ta[r,g] for g in goods if (a_set[r,g] != 0) )
+# production taxes - assumes lumpsum recycling  
+            + sum( (haskey(YM.lookup[1], (r, s)) ? YM[(r, s)] : 1) * ys0[r,s,g] * ty[r,s] for s in sectors, g in goods)
+            + sum( (haskey(YX.lookup[1], (r, s)) ? YX[(r, s)] : 1) * ys0[r,s,g] * ty[r,s] for s in sectors, g in goods)
+        )
+);
+
+#----------
+
+
+####################################
+# -- Complementarity Conditions --
+####################################
+
+# define complementarity conditions
+# note the pattern of ZPC -> primal variable  &  MCC -> dual variable (price)
+#@complementarity(cge,profit_y,Y);
+@complementarity(cge,profit_ym,YM);
+@complementarity(cge,profit_yx,YX);
+@complementarity(cge,profit_x,X);
+@complementarity(cge,profit_a,A);
+@complementarity(cge,profit_c,C);
+@complementarity(cge,profit_ms,MS);
+@complementarity(cge,market_pa,PA);
+@complementarity(cge,market_py,PY);
+@complementarity(cge,market_pd,PD);
+@complementarity(cge,market_pn,PN);
+@complementarity(cge,market_pl,PL);
+#@complementarity(cge,market_pk,PK);
+@complementarity(cge,market_rk,RK);
+@complementarity(cge,market_rkx,RKX);
+@complementarity(cge,market_pm,PM);
+@complementarity(cge,market_pc,PC);
+@complementarity(cge,market_pfx,PFX);
+@complementarity(cge,income_ra,RA);
+
+#----------
+#Recursive Dynamics
+
+
+####################
+# -- Model Solve --
+####################
+
+#set up the options for the path solver
+PATHSolver.options(convergence_tolerance=1e-6, output=:yes, time_limit=3600, cumulative_iteration_limit=0)
+
+# export the path license string to the environment
+# this is now done in the SLiDE initiation steps 
+ENV["PATH_LICENSE_STRING"]="2617827524&Courtesy&&&USR&64785&11_12_2017&1000&PATH&GEN&31_12_2020&0_0_0&5000&0_0"
+
+# solve the model
+
+status = solveMCP(cge)
+
+for t in 1:3
```

Comment:
This is good advice, thank you. We will run this for as many years into the future as we want to, so it will probably be user-specified somewhere. `length(years)` will come in handy. But agreed, this is more of just a placeholder for testing purposes. Much work needed in *_recurse.jl.

Yes, the while loop could come in handy when looping within a year to get convergence with ReEDS.